src/disk.hpp

```diff
@@ -100,20 +100,35 @@ struct FileDisk {
         Open(false);
     }
 
-    void Open(bool read_only = true)
+    void Open(bool read_only = true, bool retry = false)
     {
         // if the file is already open, don't do anything
         if (f_) return;
 
         // Opens the file for reading and writing
+        if (retry) {
+            do {
 #ifdef _WIN32
-        f_ = ::_wfopen(filename_.c_str(), read_only ? L"r+b" : L"w+b");
+                f_ = ::_wfopen(filename_.c_str(), read_only ? L"r+b" : L"w+b");
 #else
-        f_ = ::fopen(filename_.c_str(), read_only ? "r+b" : "w+b");
+                f_ = ::fopen(filename_.c_str(), read_only ? "r+b" : "w+b");
 #endif
-        if (f_ == nullptr) {
-            throw InvalidValueException(
-                "Could not open " + filename_.string() + ": " + ::strerror(errno));
+                if (f_ == nullptr) {
```

Comment:
it seems to me that this code could be simplified by *only* checking the `retry` setting right here (i.e. not branch on it earlier). If it's not set, throw here, otherwise move along

Reply:
Yeah, it's not very compact. Thanks, I'll try your suggestion.